pkg/scheduler/scheduler.go

```diff
@@ -780,11 +783,23 @@ type podConditionUpdaterImpl struct {
 
 func (p *podConditionUpdaterImpl) update(pod *v1.Pod, condition *v1.PodCondition) error {
```

Comment:
I think we need a unit test for this struct

Reply:
Definitely agree... I'll add a test around this and also for podPreempterImpl.  I'll wait until we run the integration tests 10 times to make sure this will fix the flake and then add the unit tests

Reply:
Tests added